### PR TITLE
ble_hal test cases achitectural change

### PR DESCRIPTION
**Pull request by bcukuro**
ble_hal test cases achitectural change

Description
-----------
Re-organization of ble hal test files to accommodate adding additional test groups into their own separate files.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_stress_test.c [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+TEST_TEAR_DOWN( Full_BLE_Stress_Test )
+{
+}
+
+/*-----------------------------------------------------------*/
+
+TEST_GROUP_RUNNER( Full_BLE_Stress_Test )
+{
+    int loop;
```
Comment (huguesBouvier) [inline libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_stress_test.c]: Remove int and use uintx_t

File: libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_stress_test.c [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    /* Initializations that need to be done once for all the tests. */
+    RUN_TEST_CASE( Full_BLE, BLE_Setup );
+    
+    RUN_TEST_CASE( Full_BLE_Stress_Test, BLE_Stack_Init_DeInit_Setup );
+    for ( loop = 0; loop < TOTAL_NUMBER_STRESS_TEST; loop++ )
```
Comment (huguesBouvier) [inline libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_stress_test.c]: Brackets

File: libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_stress_test.c (hunk 1 above)
Comment (bcukuro) [reply libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_stress_test.c]: changed to uinty16_t

File: libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_stress_test.c (hunk 2 above)
Comment (bcukuro) [reply libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_stress_test.c]: Brackets added

**Pull request merged**